### DISCUSSION TITLE
Reset specific localStorage key

### DISCUSSION THREAD

**Discussion by thewh1teagle** [juliencrn/usehooks-ts#642] [Help]
How can I reset local storage item?

say I have the following:

```typescript
const [state, setState] = useLocalStorage('state', 0)
```

Say I changed it to 10. and I want to reset it to default value passed to the hook. how can I reset it and reflect it in the UI?
eg. can I just do `localStorage.removeItem(...)`
I don't want to manually set the initial value but rather just reset it somehow in a way that also the state will update immediately with the initial value.

**Comment by tamish-max** [comment]
`useLocalStorage` already gives you the reset/remove behavior as the **3rd tuple value**.

```ts
const [state, setState, removeState] = useLocalStorage('state', 0)

removeState() // removes the key and immediately resets React state to the initial value
```

That is usually better than calling `localStorage.removeItem(...)` yourself, because the hook also updates its internal state for the current component.

If you remove the key manually, the browser `storage` event won’t fire in the same tab, so your UI can get out of sync.

So for your case, I’d use the built-in remover rather than manually setting the initial value again.

**Reply by thewh1teagle** [reply]
I asked it 2 years ago, back then we didn't have removeState lol... Thanks
